tareas/Miguel/Tarea1Resuelta.jl

```diff
@@ -0,0 +1,270 @@
+
+#-
+# Definimos la f y df de prueba:
+
+#- 
+f(x) = x^2 -2
+#-
+df(x) = 2x
+
+#-
+# Probamos la funcion newton con una condición inicial arbitraria x0 = 4.
+
+#-
+newton(f,df,4.0)
+
+#-
+# Corroboramos que en efecto, nos devuelve la raiz de 2, con 5 iteraciones.
+
+#-
+sqrt(2)
+
+#-
+#Con esto en mano, analicemos el número de pasos realizados, con respecto a $|x_{n-1}-x_n|$. Para ello, realizamos una nueva
+# función llamada *newton2*, donde almacenaremos en 3 arrays declarados globalmente, la raices obtenidas, el número de pasos
+# hechos y la diferencia entre cada valor iterado previo y consecutivo.
+
+#-
+raicesf = Float64[]
+Delta = Float64[]
+npasos = Float64[]
+
+#-
+function newton2(f,df, x, raiz, delta, pasos)
+    x0 = x
+    c = 0
+    x1 = 0.0
+    while abs(f(x0)) > 0.00005
+        x1 = x0 - f(x0)/df(x0)
+        push!(raiz, x1)
+        push!(delta, abs(x1 - x0))
+        x0 = x1
+        c = c+1
+        push!(pasos,c)
+    end
+    return pasos, x1, c
+end
+
+#-
+# Probamos nuestra nueva función, con f y df, y los arrays respectivos.
+
+# Nótese que ahora la función pide 3 argumentos más, los arrays van en orden respectivo:
+# - raiz es cualquier arreglo de tipo flotante para almacenar las iteraciones.
+# - delta es cualquier arreglo de tipo flotante para almacenar la diferencia entre iteraciones consecutivas.
+# - pasos es cualquier arreglo de tipo int o flotante para almecenar el número de pasos, nada que no pueda generarse por cuenta propia.
+
+#-
+newton(f,df,8.0,raicesf,Delta,npasos)
+
+#-
+# Procedemos a graficar el número de pasos vs. las delta de los valores iterados.
+
+#-
+using Plots
+
+#-
+x = npasos
+y = Delta
+scatter(x, y, title = "No. pasos vs Delta", xlabel = "Pasos", ylabel = "Delta" , label = "Delta", lw = 3)
+
+#-
+# Por lo tanto, garantizamos la convergencia del método para la funcion *f*, puesto que la diferencia se va haciendo cada vez más
+# pequeña, hasta llegar a cero.
+
+#-
+
+# ## 2
+# 
+# - Usando la función que implementaron en el ejercicio anterior y variando la condición 
+# inicial `x0` de -3 a 3 con pasos *suficientemente* pequeños, por ejemplo `0.125`, 
+# grafiquen la dependencia de la raíz encontrada en términos de la condición inicial para 
+# la función $f(x)=x^2-2$. Para hacer esto, vale la pena que guarden en un vector la raíz 
+# obtenida y en otro la condición inicial.
+# 
+# - Repitan el inciso anterior para $g(x) = (x-1)(x-2)(x-3)$, considerando el intervalo 
+# $x_0\in[1,3]$ y varias condiciones iniciales distintas. ¿Qué pueden decir de la dependencia
+# del resultado que obtienen de la condición inicial.
+# 
+
+#- 
+# ## Solución:
+#- 
+# Haremos una nueva función llamada *newton3*, donde ahora variaremos la condición inicial utilizada para el método, y pediremos
+# que nos almacene en otro nuevo arrays las raíces generadas para cada condición inicial respectivamente.
+
+#-
+cond1 = [i for i in -3:0.125:3] 
```

Comment:
No veo la necesidad de definir `newton3`; puedes usar `newton` en los distintos valores de `-3:0.125:3.0`. De hecho, si usas un `for` de manera correcta, puedes incluso evitar *crear* el arreglo `cond1`.